src/Stratis.Bitcoin.Features.Wallet/Wallet.cs

```diff
@@ -380,6 +405,47 @@ public HdAccount AddNewAccount(string password, string encryptedSeed, byte[] cha
 
             return newAccount;
         }
+
+        /// <summary>
+        /// Adds an account to the current account root using extended public key.
+        /// </summary>
+        /// <param name="accountExtPubKey">The extended public key for the account.</param>
+        /// <param name="accountIndex">The zero-based account index.</param>
+        /// <param name="network">The network for which this account will be created.</param>
+        /// <param name="accountCreationTime">Creation time of the account to be created.</param>
+        /// <returns>A new HD account.</returns>
+        public HdAccount AddNewAccount(ExtPubKey accountExtPubKey, int accountIndex, Network network, DateTimeOffset accountCreationTime)
+        {
+            ICollection<HdAccount> accounts = this.Accounts;
+
+            if (accounts.Any(x => x.HdPath.Split('/').Last() == accountIndex + "'"))
```

Comment:
Maybe you could just check `if(accounts.Any(acc => acc.Index == accountIndex))`?

Reply:
yes :) much better